src/core/diags/factory_diags.cpp

```diff
@@ -854,6 +872,12 @@ void Diags::TransmitDone(Error aError)
         }
         else
         {
+            if (!mRepeatActive && !mIsAsyncSend)
```

Comment:
If we get an error in `TransmitDone()` while `mIsAsyncSend` is in use, should we output the error (and clear `mIsAsyncSend`)?